contracts/RealmDiamond/facets/ParcelRolesRegistryFacet.sol

```diff
@@ -0,0 +1,230 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import {IERC7432} from "../../interfaces/IERC7432.sol";
+import {IERC721} from "../../interfaces/IERC721.sol";
+
+import {Modifiers, RoleData} from "../../libraries/AppStorageInstallation.sol";
+
+import {LibItems} from "../../libraries/LibItems.sol";
+
+import {LibMeta} from "../../libraries/LibMeta.sol";
+
+contract ParcelRolesRegistryFacet is Modifiers, IERC7432 {
+  uint256 public constant MAX_EXPIRATION_DATE = 90 days;
+  uint256 constant MAX_SHARES_LENGTH = 100;
+
+    event _REALM(address realm);
+
+  /** Modifiers **/
+
+    modifier onlyRealm(address _tokenAddress) {
+        emit _REALM(s.realmDiamond);
+        require(_tokenAddress == s.realmDiamond, "ParcelRolesRegistryFacet: Only Item NFTs are supported");
+        _;
+    }
+
+  /**
+   * @notice Checks if the sender is the owner or approved for all.
+   * @dev It reverts if the sender is not the owner or approved for all.
+   * @param _account The account to check.
+   * @param _tokenAddress The token address.
+   */
+  modifier onlyOwnerOrApproved(address _account, address _tokenAddress) {
+    address _sender = LibMeta.msgSender();
+    require(_account == _sender || isRoleApprovedForAll(_tokenAddress, _account, _sender), "ParcelRolesRegistryFacet: account not approved");
```

Comment:
There is a critical bug here, the user must be the owner of the NFT or approved by the owner of the NFT. Instead, you are comparing the `msg.sender` with `LibMeta.msgSender()`